bridge/ApiRequestHandler.cpp

```diff
@@ -40,358 +18,67 @@ httpd::Response makeErrorResponse(httpd::StatusCode statusCode, const std::strin
+        return ApiActions::ABOUT;
 
-const api::EndpointDescription::SsrcGroup* findFeedbackGroup(const api::EndpointDescription::Video& video,
-    const uint32_t ssrc)
-{
-    for (auto& ssrcGroup : video._ssrcGroups)
-    {
-        if (ssrcGroup._semantics.compare("FID") == 0 && ssrcGroup._ssrcs.size() == 2 && ssrcGroup._ssrcs[0] == ssrc)
-        {
-            return &ssrcGroup;
-        }
-    }
-    return nullptr;
-}
+    if (utils::StringTokenizer::isEqual(outToken, "stats") && request._method == httpd::Method::GET)
+        return ApiActions::STATS;
 
-std::vector<bridge::SimulcastStream> makeSimulcastStreams(const api::EndpointDescription::Video& video,
-    const std::string& endpointId)
-{
-    std::vector<bridge::SimulcastStream> simulcastStreams;
-    for (const auto sourcesSsrc : video._ssrcs)
+    if (utils::StringTokenizer::isEqual(outToken, "conferences"))
     {
-        auto simulcastGroup = findSimulcastGroup(video, sourcesSsrc);
-
-        if (simulcastGroup)
-        {
-            assert(simulcastGroup->_ssrcs.size() > 1);
-            const auto sources = simulcastGroup->_ssrcs;
-            if (std::find(sources.begin() + 1, sources.end(), sourcesSsrc) != sources.end())
-            {
-                continue;
-            }
-        }
-
-        if (simulcastGroup && sourcesSsrc == simulcastGroup->_ssrcs[0])
+        if (outToken.next)
         {
-            bridge::SimulcastStream simulcastStream{0};
-
-            for (auto& ssrcAttribute : video._ssrcAttributes)
+            if (request._method == httpd::Method::GET)
             {
-                if (ssrcAttribute._content.compare(api::EndpointDescription::SsrcAttribute::slidesContent) == 0 &&
-                    ssrcAttribute._ssrcs[0] == sourcesSsrc)
-                {
-                    simulcastStream._contentType = bridge::SimulcastStream::VideoContentType::SLIDES;
-                }
-            }
-
-            for (auto simulcastSsrc : simulcastGroup->_ssrcs)
-            {
-                const auto feedbackGroup = findFeedbackGroup(video, simulcastSsrc);
-                if (!feedbackGroup)
-                {
-                    continue;
-                }
-
-                simulcastStream._levels[simulcastStream._numLevels]._ssrc = simulcastSsrc;
-                simulcastStream._levels[simulcastStream._numLevels]._feedbackSsrc = feedbackGroup->_ssrcs[1];
-                ++simulcastStream._numLevels;
-
-                logger::debug("Add simulcast level main ssrc %u feedback ssrc %u, content %s, endpointId %s",
-                    "ApiRequestHandler",
-                    simulcastSsrc,
-                    feedbackGroup->_ssrcs[1],
-                    toString(simulcastStream._contentType),
-                    endpointId.c_str());
+                auto nextToken = utils::StringTokenizer::tokenize(outToken, '/');
+                if (nextToken.next)
+                    return ApiActions::GET_ENDPOINT_INFO;
```

Comment:
We could have just made the call to the actionHandler here. I do not see it would make the structure more complicated.

Reply:
Agree!